proto/api.go

```diff
@@ -86,6 +86,12 @@ type Request interface {
 	flags() int
 }
 
+// RequestWithUser is an interface for RPC requests that have a "requested user".
+type RequestWithUser interface {
```

Comment:
I thought about that but figured it would be clearer to have it in the same place as Request and its implementers. I'm happy moving it though if you insist.